File: Code Forces/C++/767A.cpp
#include <stdio.h>
 
bool snacks[100009];
 
int main()
{
    int n, a, next;
    scanf_s("%d", &n);
    next = n;
    for (int i = 0; i < n; i++) {
        scanf_s("%d", &a);
        snacks[a] = true;
        if (a == next) {
            printf("%d ", a);
            next--;
            snacks[a] = false;
            for (int j = a - 1; j >= 1 && snacks[j]; j--) {
                if (snacks[j]) {
                    printf("%d ", j);
                    next = j - 1;
                    snacks[j] = false;
                }
            }
        }
        printf("\n");
    }
}

File: Code Forces/Python/767A.py
n = int(input())
next = n
snacks = [False] * (n+2)
orderedSnacks = [0]*(n+2)
 
orderedSnacks = list(input().split(' '))
for i in range(0, n):
    orderedSnacks[i] = int(orderedSnacks[i])
 
for i in range(1, n+1):
    inSnack = orderedSnacks[i-1]
    snacks[inSnack] = True
    if(inSnack == next):
        snacks[next] = False
        print(next, ' ', sep='', end='')
        next = next - 1
        for next in range(next, 0, -1):
            if(snacks[next] == True):
                print(next, ' ', end='')
                next = next - 1
            else:
                break
    print('')
